### PR TITLE
feat(pagination): repeat table headers

### PR DESCRIPTION
**Pull request by vsevolod**
## Summary

Adds support for repeating table headers across page breaks. Header rows with computed `display: table-header-group` are repeated on continuation pages, with space reserved before body-row pagination.

## Changes

- Detect repeating table header groups from computed display values.
- Repeat complete header bands on continuation pages.
- Reserve header space when paginating table body rows.
- Preserve correct geometry, clipping, links, and tagged PDF structure.
- Handle nested tables, multi-row and multiple header groups, oversized headers, mid-page table starts, and bounded clone budgets.
- Size rendered split-table fragments from their fragment geometry.
- Add unit, integration, regression, and visual tests.

## Examples to verify
- HTML [repeating-table-headers-demo.html](https://github.com/user-attachments/files/31037381/repeating-table-headers-demo.html)

- Generated PDF [repeating-table-headers-demo.pdf](https://github.com/user-attachments/files/31037363/repeating-table-headers-demo.pdf)

## Test plan
- [x] `cargo test -p fulgur`
- [x] `cargo clippy`
- [x] `cargo fmt --check`
- [x] `npx markdownlint-cli2 '**/*.md'` (not applicable; docs unchanged)

## Contributor License Agreement

By opening this pull request, I confirm that:

- [x] I have read the [Contributing Guide](https://github.com/fulgur-rs/fulgur/blob/main/CONTRIBUTING.md)
- [x] I have read the [CLA](https://github.com/fulgur-rs/fulgur/blob/main/CLA.md) and agree to its terms. (First-time contributors: the CLA Assistant bot will comment on this PR with sign-in instructions.)

## 新機能
- 複数ページにまたがる表で、ヘッダーを各ページに自動表示します。
- 複数行・ネストした表や複雑なレイアウトのページ分割を改善しました。
- 表の高さ、枠線、クリッピングをページごとに適切に処理します。
- ヘッダー内のリンクやタグ付きPDFの情報を各ページで保持します。

## テスト
- 長い表、リンク注釈、PDF出力、境界条件の検証を追加しました。

## ドキュメント
- 表ヘッダーの繰り返し仕様、無効化方法、検証用サンプルを追加しました。

### CODE REVIEW

**Comment by vsevolod** [comment]
I have read the CLA Document and I hereby sign the CLA

File: crates/fulgur/src/render.rs [hunk 1]
```diff
@@ -2835,48 +2835,54 @@ fn draw_block_inner_paint(
...
+            .map(|s| s.height.to_f32())
+            .unwrap_or_else(|| {
+                let from_frag = fragment_height;
+                if from_frag > 0.0 {
+                    from_frag
+                } else {
+                    entry.cached_height.to_f32()
+                }
+            })
+    };
```
[the remark this answers is not in the document]
Comment (vsevolod) [reply crates/fulgur/src/render.rs:2894]: The zero-height page-0 fragment is real — `nested_oversized_header_fallback_honors_occupied_outer_strip` is not a vacuous `.all()`. For that fragment `table_box_size` misses the `is_split && height > 0` arm, and `draw_table_v2` / `draw_under_clip_table` still paint when `visible`.

I reproduced it with a bordered/background inner table: page 1 leaks a sliver of the inner table's top border on the leftover strip. It is not a full-height ghost frame. With this PR's fixture (no table chrome) `paint_table_outer_frame` is a no-op, so nothing is visible.

Not changing this here:

- The skip belongs in `draw_table_v2` / `draw_under_clip_table` for split fragments with `frag.height <= 0`, not in `table_box_size`. Unsplit zero-height fragments must still fall back to `cached_height` (`table_box_size_no_layout_size_zero_frag_falls_back_to_cached_height`).
- `nested_oversized_header_fallback_honors_occupied_outer_strip` already exists and only asserts geometry. A useful lock would be a render test with visible table chrome, not another copy of that pagination test.

Follow-up if we want the chrome leak closed; not a blocker for repeating headers.

**Comment by vsevolod** [comment]
PR description updated: Renew HTML and generated PDF. Added more edge cases

**Comment by mitsuru** [comment]
@vsevolod 
Thanks for taking this on — context first, since it explains the state of this area.

Repeating table headers worked in fulgur's earlier (Pageable-era) architecture.
We lost them in the re-architecture and deliberately deferred restoring them
rather than block the migration. So this recovers something we knowingly left
behind, and I appreciate you picking it up.

Reviewing it, I ran some edge cases through `Engine::layout()`. Worth flagging:
**nested tables don't paginate.** A table inside a cell of another, `<thead>` on
both levels, 200×100 CSS px page:

```
outer table  = [(page 0, y=0,   h=182)]   single fragment on a 100px page
inner header = [(page 0, y=20,  h=18)]    never repeats
last row     = [(page 0, y=158, h=24)]    bottom 182 — overflows by 82px
```

I reproduced this on your branch head with nothing applied, so please don't
treat it as this PR's to solve.

It's upstream, and it shares a root with the other rough edge here: Taffy has no
table layout algorithm — `taffy::compute` ships `block`, `flexbox`, `grid` and
`leaf`, and tables are approximated on those. Nested tables and cell alignment
are where the approximation shows. The related gap is that a
`table-header-group` isn't hoisted to the top of its table per CSS 2.1 §17.5.1
(WPT `css/CSS2/tables/table-header-group-005.xht` fails for this, and did before
this PR). That's the wall we hit during the re-architecture.

What's changed is my read of how much fulgur can do on its own side — more than
I assumed when we deferred it, and I'm looking into that. My thread to pull, not
a request on this PR.

Detailed review to follow; nothing above is a blocker.

**Comment by mitsuru** [comment]
@vsevolod Merged — thank you. Repeating table headers are back in `main`.

It landed through #721, a follow-up branch carrying your commits plus fixes from
the review, so GitHub marked this PR merged automatically. Your four commits are
in the history attributed to you, not squashed.

What #721 changed on top:

- Header groups not at the top of their table no longer repeat — one placed
  after a `tbody` reserved a band spanning the rows above it, giving a fragment
  taller than the page. Also applies css-tables-3's "only the first
  `table-header-group` is a header".
- **A table no longer starts on a page where only the header fits.** This is a
  deliberate departure from your demo, which called for page 2 to show the
  header alone. No normative test settles it — CSS 2.1's
  `table-header-group-004` is flagged `may paged` and passes either way — so we
  read the intent off the repeated-section tests in
  `css/css-break/table/repeated-section/`, which reference css-tables-3
  §repeated-headers. A header stranded above nothing costs a sheet to convey
  nothing and gets redrawn overleaf regardless. Note we still don't push a row
  that couldn't fit on an empty page either, matching
  `tall-monolithic-after-repeated-header`.
- Fixed a row torn across pages: `start_initial_page_occupied` is per-row, but
  was cleared mid-row, so later cells skipped the page-break check.
- A header band containing a forced break is no longer repeated — the table
  paginates normally instead and the break is honoured.
- Plus trailing forced breaks, out-of-flow children in the reserve, and painted
  zero-height boxes losing their table slice.

Two things for next time:

**Goldens need the font pinning.** This one was generated without
`FONTCONFIG_FILE`, so it captured your system fonts instead of the bundled Noto
Sans and didn't match CI:

```
FONTCONFIG_FILE="$PWD/examples/.fontconfig/fonts.conf" \
FULGUR_VRT_UPDATE=failing cargo test -p fulgur-vrt
```

**Fork PRs don't run our full CI** — no test, lint or VRT jobs fired here, which
is why that mismatch went unnoticed until #721. Ours to fix, not yours.

Remaining limitations are documented in `docs/css-support.md` (nested tables
don't fragment, header groups aren't hoisted per CSS 2.1 §17.5.1, forced breaks
on rows aren't honoured). Most trace back to Taffy having no table layout
algorithm, so they're upstream rather than anything this PR introduced.

Thanks again for picking up something we'd deliberately shelved.
